lib/util/Text.js

```diff
@@ -133,21 +134,23 @@ function fit(lines, fitLine, originalLine, textBBox) {
  * @return {string} the shortened string
  */
 function semanticShorten(line, maxLength) {
-  var parts = line.split(/(\s|-)/g),
+
+  var parts = line.split(/(\s|-|­)/g),
```

Comment:
The regex doesn't look like it contains a special character at the end, it could inadvertently be removed by a future maintainer. I think this line deserves a comment that warns that a soft hyphen is present in the regex.
Or alternatively, you can match it using the unicode value `\u00AD`.

Reply:
Agreed. Good catch.

Reply:
Using explicit unicode char now. Much better: https://github.com/bpmn-io/diagram-js/pull/508/commits/3981c21ae4c79ca47526636376cd1c6e13f84774